### PR TITLE
[wasm][bindings] Add bindings for JavaScript Core Map object.

### PR DESCRIPTION
**Pull request by kjpou1**
- The Map object is more or less the equivalent of a C# HashTable.
- Implemented the IDictionary interface within the object so it can be treated like a managed object.
- Note that the Map iterator is non resettable and can only be iterated once.
   - The Reset() method on the Enumerators are implemented to dispose of the current MapIterator and retrieve a new from the map so iterating over the collections multiple times function correctly.
- Tests added. 

examples:

```

            var myMap = new Map();
            myMap["hello"] = "world";
            myMap["foo"] = "worldfoo";

            Console.WriteLine(myMap.Count);
            Console.WriteLine(myMap.Contains("hello"));
            foreach (DictionaryEntry item in myMap)
            {
                Console.WriteLine($"Key: {item.Key} Value: {item.Value}");
            }
            foreach (object key in myMap.Keys)
            {
                Console.WriteLine($"Key: {key} Value from key: {myMap[key]}");
            }

            foreach (object val in myMap.Values)
            {
                Console.WriteLine($"Value: {val}");
            }

```

output:

```
2
True
Key: hello Value: world
Key: foo Value: worldfoo
Key: hello Value from key: world
Key: foo Value from key: worldfoo
Value: world
Value: worldfoo
```

Note: 
- Map allows null keys
- Map does not check for duplicate key on `Add` it will just replace the value with the value being added.

### CODE REVIEW

**Comment by lewing** [comment]
@monojenkins build failed

**Comment by kjpou1** [comment]
@monojenkins build failed

**Comment by lewing** [comment]
@monojenkins build failed

File: sdks/wasm/binding_support.js [hunk 1]
```diff
@@ -823,14 +827,20 @@ var BindingSupportLib = {
 							SharedArrayBuffer.prototype[Symbol.for("wasm type")] = coreType
 						}
 						break;
+					case "Map":
+						coreType = this.wasm_get_core_type(obj);
+						if (typeof coreType !== "undefined") {
+							Map.prototype[Symbol.for("wasm type")] = coreType
+            }
```
Comment (lewing) [inline sdks/wasm/binding_support.js]: nit: whitespace

File: sdks/wasm/framework/src/WebAssembly.Bindings/Core/Map.cs [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+
+		}
+
+		public int Count => (int)GetObjectProperty ("size");
+
+		public bool IsSynchronized => false;
+
+		public object SyncRoot => false;
+
+		public void Add (object key, object value)
```
Comment (lewing) [inline sdks/wasm/framework/src/WebAssembly.Bindings/Core/Map.cs]:
```suggestion
		public void Add (object key, object value) => Invoke ("set", key, value);
```

**Comment by kjpou1** [comment]
@monojenkins build failed

**Comment by kjpou1** [comment]
@monojenkins build failed

**Comment by kjpou1** [comment]
@monojenkins build failed

**Comment by kjpou1** [comment]
@monojenkins build Linux AArch64

**Comment by lewing** [comment]
@monojenkins build failed

**Comment by kjpou1** [comment]
@monojenkins build OS X x64 Android SDK

**Comment by kjpou1** [comment]
@monojenkins build OS X x64 Android SDK

**Comment by kjpou1** [comment]
@monojenkins build failed

**Comment by kjpou1** [comment]
@monojenkins build failed

**Comment by kjpou1** [comment]
@monojenkins build failed

**Comment by kjpou1** [comment]
@monojenkins build failed

**Comment by kjpou1** [comment]
@monojenkins build OS X x64 Hardened Runtime

**Comment by lewing** [comment]
closing this because of the conflicts, feel free to reopen once they are resolved

**Comment by kjpou1** [comment]
Rebased PR on current master.

**Pull request closed without merging**